service/lib/agama/storage/space_settings.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+# Copyright (c) [2023] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+module Agama
+  module Storage
+    # Settings regarding the Agama storage proposal for making space
+    class SpaceSettings
+      POLICIES = [:delete, :keep, :resize, :custom].freeze
+      private_constant :POLICIES
+
+      # @return [Symbol] :delete, :keep, :resize or :custom
+      attr_accessor :policy
+
+      # @return [Hash] used only for :custom
```

Comment:
btw I think there is even design pattern that suggest to use object and inheritance for different kind of policy. But I think here it is overkill. But if you plan to extend it in future, then it is something to consider.

Reply:
Yes. In fact, @ancorgs and me have already discussed about using some objects for the size config too (`AutoSize` vs `ManualSize`). This here is a similar case. For now we decided to keep this simple approach and improve it later if needed.